src/huggingface_hub/hf_api.py

```diff
@@ -1177,18 +1177,55 @@ def delete_repo(
 
     def update_repo_visibility(
         self,
-        name: str,
-        private: bool,
+        name: str = None,
+        private: bool = False,
         token: Optional[str] = None,
         organization: Optional[str] = None,
         repo_type: Optional[str] = None,
+        repo_id: str = None,
     ) -> Dict[str, bool]:
-        """
-        Update the visibility setting of a repository.
+        """Update the visibility setting of a repository.
+
+        Args:
+            repo_id: A user or an organization name and a repo name seperated by a
+                ``/``.
+
+                .. versionadded: 0.4.0
+
+            private: Whether the model repo should be private (requires a paid
+                huggingface.co account)
+
+            token: A token to be used for the download.
+                - If ``True``, the token is read from the HuggingFace config
+                folder.
+                - If a string, it's used as the authentication token.
```

Comment:
`token` should always be a string, so I would keep that. If not specified, it retrieves the stored token.

Reply:
`_validate_or_retrieve_token` seems to disagree with you I think. Although in this case you're right I guess? I'm not sure if we're being consistent in treatment of `token` and `use_auth_token`  across this file.